plugins/scaffolder/src/types.ts

```diff
@@ -64,3 +64,16 @@ export type ListActionsResponse = Array<{
     output?: JSONSchema;
   };
 }>;
+
+type OutputLink = {
+  url: string;
+  title?: string;
+  icon?: string;
+};
+
+export type TaskOutput = {
+  entityRef?: string;
+  remoteUrl?: string;
+  links?: OutputLink[];
+  [key: string]: any;
```

Comment:
I've invalidated earlier comments here so calling out that this relaxes the constraint (previously the remaining properties would be strings). I may be wrong but I don't think there's a way to express in Typescript the concept of "an object with one attribute which is type T and any number of others of type U". `[key: string]: any` works because `string | undefined` and `OutputLinks[] | undefined` are instances of `any`.

Reply:
This should be what you want

```ts
TaskOutput = {
  entityRef?: string;
  remoteUrl?: string;
  links?: OutputLink[];
} & {
  [key: string]: any;
}
```

Probably better to use `unknown` over `any` here tho